wire_test.go

```diff
@@ -206,6 +207,86 @@ func TestServerWritingResult(t *testing.T) {
 	})
 }
 
+func TestServerHandlingMultipleConnections(t *testing.T) {
+	address := TConstructMockPsqlServer(t)
+	databaseUrl := fmt.Sprintf("postgres://%s:%d", address.IP, address.Port)
+	conn, err := sql.Open("pgx", databaseUrl)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		_ = conn.Close()
+	})
+
+	t.Run("test connectivity", func(t *testing.T) {
+		err := conn.Ping()
+		require.NoError(t, err)
+	})
+
+	t.Run("simple query execution should return rows", func(t *testing.T) {
+		rows, err := conn.Query("select age from person")
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			rows.Close()
+		})
+		require.True(t, rows.Next())
+		require.NoError(t, rows.Err())
+	})
+
+	t.Run("prepared statement query execution should support all parameter styles", func(t *testing.T) {
+		type input struct {
+			query string
+		}
+		scenarios := map[string]input{
+			"dollar param style": {
+				query: "select age from person where age > $1",
+			},
+			"question mark param style": {
+				query: "select age from person where age > ?",
+			},
+		}
+		for name, testInput := range scenarios {
+			query := testInput.query
+			t.Run(name, func(t *testing.T) {
+				stmt, err := conn.Prepare(query)
+				require.NoError(t, err)
+				t.Cleanup(func() {
+					stmt.Close()
+				})
+				rows, err := stmt.Query(1)
+				require.NoError(t, err)
+				t.Cleanup(func() {
+					rows.Close()
+				})
+				require.True(t, rows.Next())
+				require.NoError(t, rows.Err())
+			})
+		}
+	})
+}
+
+func TConstructMockPsqlServer(t *testing.T) *net.TCPAddr {
+	t.Helper()
+	handler := func(ctx context.Context, query string, writer DataWriter, parameters []string) error {
+		t.Log("serving query")
+		writer.Define(Columns{ //nolint:errcheck
+			{
+				Table:  0,
+				Name:   "age",
+				Oid:    oid.T_int4,
+				Width:  1,
+				Format: TextFormat,
+			},
+		})
+		writer.Row([]any{20}) //nolint:errcheck
+		return writer.Complete("OK")
+	}
+	d, err := zap.NewDevelopment()
+	require.NoError(t, err)
```

Comment:
I would like to suggest to use the zap [zaptest](https://pkg.go.dev/go.uber.org/zap/zaptest#NewLogger) package instead of using a development logger.

Reply:
Done